core/ast/src/scope.rs

```diff
@@ -105,6 +105,11 @@ pub(crate) struct Inner {
     function: bool,
     // Has the `this` been accessed/escaped outside the function environment boundary.
     this_escaped: Cell<bool>,
+
+    /// A counter for unique IDs for Scopes generated as part of this scope tree.
+    ///
+    /// Only used in the root scope.
+    unique_id_ctr: Cell<u32>,
```

Comment:
Suggestion: It might be cleaner to keep the Scope state separate for example, by introducing a `ScopeContext` that holds this data. It could then be passed into Scope::new, or each Scope could store an Rc<ScopeContext> (eliminates the root search for next unique id)